types/graphql/type/definition.d.ts

```diff
@@ -262,9 +262,9 @@ export type GraphQLIsTypeOfFn<TSource, TContext> = (
     info: GraphQLResolveInfo
 ) => boolean | Promise<boolean>;
 
-export type GraphQLFieldResolver<TSource, TContext> = (
+export type GraphQLFieldResolver<TSource, TContext, TArgs = any> = (
```

Comment:
This is weakening the existing typings. Anyone who uses GraphQLFieldResolver with 2 type arguments (which is done further down in this file) will get the default `TArgs = any`.
This will cause `args` to be instantiated as `any & { [argName: string]: any }` which is equivalent to just `any`.

Reply:
I'm not familiar with the library so not totally sure how to advise on achieving your goal, but adding a test case might help you come up with a better solution.

Reply:
It's less of an issue now that the graphql-tools library made a [change](https://github.com/apollographql/graphql-tools/commit/fa4f5b3046e7541e7036392faf66a6ff1638a4cb#diff-3f45022c70dd5ce8f6c044da93286d6cL32) to move away from using `GraphQLFieldResolver` from graphql and instead use it's own `IFieldResolver`.

This goal of this change is an effort to allow for a typed `args` object. Ignoring the specifics of the graphql library, I'd be appreciative and interested to hear any recommendations you have for how to accomplish this. For instance, I have the following that throws a TS error stating `Property 'id' is missing in type '{ [argument: string]: any; }'`.

```ts
getTagById: (source: any, args: { id: number }) => {
  // ...
}
```

Considering `IFieldResolver` looks like the following, I would think having an indexer property for the `args` parameter doesn't seem better than `any`, while using the intersection type allows the project to compile successfully and makes it possible to specify the shape of the `args`.

```ts
export declare type IFieldResolver<TSource, TContext> = (
  source: TSource,
  args: {
    [argument: string]: any;
  },
  context: TContext,
  info: GraphQLResolveInfo & {
    mergeInfo: MergeInfo;
  },
) => any;
```

Reply:
have you considered making it `object` instead